src/Checkers/LicenseChecker.js

```diff
@@ -0,0 +1,208 @@
+  }
+
+  check(dirpath) {
+    const files = this._getFiles(dirpath, []);
+    const errors = [];
+    for (const i in files) {
+      const parsedPath = path.parse(files[i]);
+      if (this._extensionsSet.has(parsedPath.ext)) {
+        errors.push(this._checkSourceFile(files[i]));
+      } else if (parsedPath.name == 'LICENSE') {
+          errors.push(this._checkLicenseFile(files[i]));
+      }
+    }
+    return errors.filter((error) => error != null);
+  }
+
+  _checkLicenseFile(filepath) {
+    const content = fs.readFileSync(filepath, 'utf-8');
+    const originalLicense = fs.readFileSync(LICENSE_FILE_PATH, 'utf-8');
+    const checkedLicense = content.replace(/\d\d\d\d(\-\d\d\d\d)?/, '');
+    return this._compareLicenses(checkedLicense, originalLicense);
+  }
+
+  _checkSourceFile(filepath) {
+    const content = fs.readFileSync(filepath, 'utf-8');
+    const lines = content.split(/\n/);
+    if (!lines) {
+      return new LicenseCheckerWarning('File have not License Header', 0, filepath);
+    }
+
+    const licenseLines = [];
+    let commentsType = '';
+    let offset = 0;
+    for (let i = 0; i < lines.length; i++) {
+      lines[i] = lines[i].trim();
+
+      if (commentsType == '') {
+
+        if (lines[i] == '' || lines[i].startsWith('#!')) {
+          offset++;
+          continue;
+        } // skip empty and shebang strings
+
+        if (lines[i].startsWith('//')) {
+          commentsType = '//';
+          licenseLines.push(lines[i].substring(2).trim());
+        } else if (lines[i].startsWith('/*')) {
+          commentsType = '/*';
+          licenseLines.push(lines[i].substring(2).trim());
+        } else {
+          return new LicenseCheckerWarning('License should be in header', i + 1, filepath);
+        }
+
+      } else if (commentsType = '//') {
+
+        if (lines[i].startsWith('//')) {
+          licenseLines.push(lines[i].substring(2).trim());
+        } else {
+          break; // end
+        }
+
+      } else if (commentsType = '/*') {
+
+        let index;
+        if (index = lines[i].indexOf('*/') > -1) {
+          licenseLines.push(lines[i].substring(0, index).trim());
+          break;
+        }
+        if (lines[i].startsWith('*')) {
+          lines[i] = lines[i].substring(1).trim();
+        }
+        licenseLines.push(lines[i]);
+
+      }
+    }
+    const checkedLicense = licenseLines.reduce((prev, curr) => prev + '\n' + curr);
+    const output = this._compareWithLicense(checkedLicense, offset);
+
+    if (output) {
+      return new LicenseCheckerWarning(output.message, offset + output.line, filepath, output.symbol);
+    }
+
+    return null;
+  }
+
+  _compareWithLicense(checkedLicense) {
+    const originalLicense = fs.readFileSync(LICENSE_PATH, 'utf-8');
+    checkedLicense = checkedLicense.replace(/\d\d\d\d(\-\d\d\d\d)?/, '');
+    return this._compareLicenses(checkedLicense, originalLicense);
+  }
+
+  _compareLicenses(checkedLicense, originalLicense) {
```

Comment:
Looks too complex as for me. May we use here something like https://www.npmjs.com/package/bloom-text-compare/